workspaces/default/themes/base/assets/js/kong/kong.utils.js

```diff
@@ -64,3 +64,196 @@ window.Kong.Utils.getErrorMessage = function (response) {
     return fallbackErrorMessage;
   }
 };
+
+/**
+ * Get value of specified nested property of an object or object array.
+ *
+ * The `key` argument accepts dot characters (`.`) to deeply traverse input objects
+ * and asterisk characters (`*`) to select all properties of currently selected object.
+ * You can use the asterisk character at any point of the object tree.
+ *
+ * Example:
+ *
+ *   For given input object:
+ *   ```
+ *   {
+ *     info: {
+ *       title: "Hello, World!",
+ *     },
+ *     tags: {
+ *       first: {
+ *         name: "First tag",
+ *       },
+ *       second: {
+ *         name: "Second tag",
+ *       },
+ *     },
+ *   }
+ *   ```
+ *   `info.title` key will get the value of the `title` property in the `info` object, and
+ *
+ *   `tags.*.name` key will get the values of the `name` property in all object properties inside the `tags` object.
+ *
+ * @param {Object|Object[]} data - An object or array of objects to extract the values from
+ * @param {string} key - Key to get the properties from given object(s)
+ * @returns {string[]} Values of requested properties
+ */
+window.Kong.Utils.getNestedProperties = function (data, key) {
+  if (!Array.isArray(data)) {
+    data = [data];
+  }
+
+  var tree = key.split(".");
+  var currentKey = tree.shift();
+  var newObjs;
+
+  if (currentKey === "*") {
+    // Find all keys of given object if current key is *
+    newObjs = data.reduce(function (acc, obj) {
+      Object.keys(obj).forEach(function (key) {
+        acc.push(obj[key]);
+      })
+      return acc;
+    }, []);
+  } else {
+    // Select the requested key from all input objects
+    newObjs = data.map(function (obj) {
+      return obj[currentKey];
+    });
+  }
+
+  if (tree.length > 0) {
+    // Recursively process nested objects
+    return window.Kong.Utils.getNestedProperties(
+      newObjs,
+      tree.join("."),
+    );
+  }
+
+  return newObjs;
+}
+
+/**
+ * Get a numeric score of string similarity using n-grams
+ * @param {string} a - first string to compare
+ * @param {string} b - second string to compare
+ * @param {number} [nGram = 3] - integer gram size (defaults to 3)
+ * @returns {number} String similarity score from 0.0 to 1.0
+ */
+window.Kong.Utils.getStringSimilarity = function (a, b, nGram) {
+  if (!nGram) {
+    nGram = 3;
+  }
+
+  if (a.length === 0 || b.length === 0) {
+    return 0;
+  }
+
+  if (b.length > a.length) {
+    var temp = a
+    a = b
+    b = temp
+  }
+
+  function getNGramPairs (str) {
+    var padding = '';
+    for (var i = 0; i < nGram - 1; i++) {
+      padding += ' ';
+    }
+
+    str = padding + str.toLowerCase() + padding;
+
+    var res = [];
+    for (var i = 0; i < str.length - nGram + 1; i++) {
+      var s = str.substring(i, i + nGram)
+      if (res.indexOf(s) === -1) {
+        res.push(s);
+      }
+    }
+
+    return res;
+  }
+
+  var aPairs = getNGramPairs(a, nGram);
```

Comment:
You are passing `nGram` but function does not accept that argument

Reply:
it's taken from the parent function (`getStringSimilarity = function (a, b, nGram)`) as `getNGramPairs` is defined inside it. The nGram variable is read-only and is safe to not pass it like this

Reply:
Yeah, I saw, In that case, we can easily change `getNGramPairs` to only accept one argument 👍🏻